src/server/object_services/object_server.js

```diff
@@ -1160,6 +1322,7 @@ function get_object_info(md) {
         upload_size: _.isNumber(md.upload_size) ? md.upload_size : undefined,
         num_parts: md.num_parts,
         version_id: bucket.versioning === 'DISABLED' ? undefined : MDStore.instance().get_object_version_id(md),
+        lock_settings: options.role === 'admin' ? md.lock_settings : undefined,
```

Comment:
if we are using options only for role why not just passing admin_role for example and pass true only when needed?

Reply:
The reason behind it was to let other people in the future use options instead of sending more and more params to the get_object_info function, AFAIK long params list is a bad smell. I agree with your suggestion if you think it probably won't be the case in here :)